### PR TITLE
Added docstrings, docstring examples and fixed typehints to linear algebra inner Close #20925

### LINKED ISSUE #20925: inner

### PR DESCRIPTION
**Pull request by AnnaBujniewicz**
Close #20925

### CODE REVIEW

**Comment by AnnaBujniewicz** [comment]
add_reformatting_checklist_linear_algebra

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/data_classes/array/linear_algebra.py [hunk 1]
```diff
@@ -361,6 +361,60 @@ def inner(
...
+            a two-dimensional array containing the inner product and whose
+            shape is (N, M).
+            The returned array must have a data type determined by Type Promotion Rules.
+
+        Examples
+        --------
+        Matrices of identical shapes
+        >>> x = ivy.array([[1., 2.], [3., 4.]])
+        >>> y = ivy.array([[5., 6.], [7., 8.]])
+        >>> d = ivy.inner(x, y)
```
Comment (Sam-Armstrong) [inline ivy/data_classes/array/linear_algebra.py:395]: These examples should be x.inner(y) not ivy.inner(x, y)

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Thanks for making the PR @AnnaBujniewicz! I've made some minor changes, please could you address the comment I made, then we should be good to merge 🙂

File: ivy/data_classes/array/linear_algebra.py (hunk 1 above)
Comment (AnnaBujniewicz) [reply ivy/data_classes/array/linear_algebra.py]: Thank you for your review! :) I applied the suggested changes now.

**Review by Sam-Armstrong — APPROVED** [review]
lgtm! thanks @AnnaBujniewicz 😁

**Pull request merged**